org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/publisher/AIProcessor.java

```diff
@@ -0,0 +1,429 @@
+      String md;
+      if (p != null) {
+        XhtmlNode xhtml = new XhtmlParser().setMustBeWellFormed(false).parse(new FileInputStream(p.f), "html");
+        switch (r.resource.asString("resourceType")) {
+          case "CodeSystem":
+            stripDiv(xhtml, "defines the following code");
+            break;
+          case "ValueSet":
+            stripDiv(xhtml, "Include these");
+            stripDiv(xhtml, "Expansion performed internally based on");
+            stripDiv(xhtml, "Expansion based on");
+            break;
+          case "StructureDefinition":
+            stripDiv(xhtml, "Differential Table");
+          default:
+            stripDiv(xhtml, "Generated Narrative: ");
+            break;
+        }
+        XhtmlToMarkdownConverter conv = new XhtmlToMarkdownConverter(false);
+        conv.configureForIGs(true);
+        md = conv.convert(xhtml);
+      } else {
+        md = "# Resource " + r.name + "\r\n\r\n";
+      }
+      if ("Binary".equals(r.resource.asString("resourceType"))) {
+        if (isText(r.resource.asString("contentType"))) {
+          md = md + "\r\n\r\n## Resource Binary Content\r\n\r\n" +
+                  r.resource.asString("contentType") + ":\r\n\r\n```\r\n" +
+                  Base64.decodeBase64(r.resource.asString("data")) +
+                  "\r\n```\r\n";
+        } else {
+          md = md + "\r\n\r\n## Resource Binary Content\r\n\r\n" +
+                  r.resource.asString("contentType") + ":\r\n\r\n```\r\n" +
+                  "{snip}" +
+                  "\r\n```\r\n";
+        }
+
+      } else {
+        md = md + "\r\n\r\n## Resource Content\r\n\r\n```json\r\n" +
+                JsonParser.compose(r.resource, true) +
+                "\r\n```\r\n";
+      }
+      if (r.filename == null) {
+        r.filename = r.resource.asString("resourceType") + "-" + r.resource.asString("id") + ".html";
+      }
+      String dest = FileUtilities.changeFileExt(r.filename, ".md");
+      if (newML) {
+        llms.append("* [" + r.name + "](" + dest.replace(".ai", "")+ ")\r\n");
+      } else {
+        llms.append("* [" + r.name + "](" + dest + ")\r\n");
+      }
+      FileUtilities.stringToFileIfDifferent(md, Utilities.path(dir, dest));
+      zip.addFileSource(dest, md, true);
+      return md.length();
+    } catch (Exception e) {
+      throw new FHIRException("Unable to process " + p.f.getAbsolutePath(), e);
+    }
+  }
+
+  private boolean isText(String contentType) {
+    if (contentType.startsWith("text/")) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private void stripDiv(XhtmlNode x, String s) {
+    List<XhtmlNode> toDelete = new ArrayList<>();
+    for (XhtmlNode c : x.getChildNodes()) {
+      if ("div".equals(c.getName()) && hasTextNotInDiv(c, s)) {
+        toDelete.add(c);
+      } else {
+        stripDiv(c, s);
+      }
+    }
+    x.getChildNodes().removeAll(toDelete);
+  }
+
+  private boolean hasTextNotInDiv(XhtmlNode c, String s) {
```

Comment:
The logic here seems to skip the top level div if I'm not mistaken. Should there not be a check for `c` as well?